### PR TITLE
Esnext modules

### PR DESCRIPTION
**Pull request by chrisprice**
Definitely not just trying to get this in before your bar/column changes. No sir. Definitely didn't cross my mind. What changes?

This is on top of #550, which failed to build so I'll fix that then rebase this one again.

### CODE REVIEW

**Comment by chrisprice** [comment]
Fixed.

**Comment by ColinEberhardt** [comment]
> What changes?

Just one or two!

One minor thing (probably not worth changing):

    import _dataJoin from '../util/dataJoin';

If I was importing a module and found it collided with the name I wanted to use for a variable, I'd probably pick a different name for the variable rather than change the name I used for the imported module.

File: src/data/data.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import coinbase from './feed/coinbase';
```
Comment (ColinEberhardt) [inline src/data/data.js]: Is it worth splitting this into `feed.js` and `random.js` under their respective folders? My first thought was 'why is this file structured differently?'

Comment (chrisprice) [inline src/data/data.js]: Laziness... fixed!

**Comment by chrisprice** [comment]
On the naming issue, I'd like to see us have a standard for it e.g. -
```
import {util_dataJoin} from '../util/dataJoin';

var dataJoin = util_dataJoin()
  //...
```

I don't like to have to think about it and it would be good to have some context when looking at whether something is a factory or an instance. 

That said, I don't think it's necessary to do in this PR. We could just have a rule to fix it up any time a file is touched?

**Comment by ColinEberhardt** [comment]
> That said, I don't think it's necessary to do in this PR. We could just have a rule to fix it up any time a file is touched?

Agreed - it's not that important right now.

**Pull request merged**